### ISSUE TITLE
const enums are not inlined

### ISSUE THREAD

**Issue by axelo** [wmonk/create-react-app-typescript#400]
### Is this a bug report?

Yes.

const enums are not inlined when building (npm build / yarn build).

The reason can be understood here https://github.com/Microsoft/TypeScript/issues/11202

And the problem/feature is this line https://github.com/wmonk/create-react-app-typescript/blob/master/config/webpack.config.prod.js#L185

Our use case: 
A large const enum that is generated and only a few values are actually used resulting in our bundle size to be a lot bigger than expected (+ 40kb). Plus we would like to think the values used to be just as strings while running and not accessed through an object.

### Can you also reproduce the problem with npm 4.x?

Didn't seem relevant.

### Which terms did you search for in User Guide?

const, enum, transpile

### Environment

1. `npm ls react-scripts-ts` (if you haven’t ejected): 
react-scripts-ts@2.17.0

2. `node -v`: 
v8.11.1

3. `npm -v`:
5.6.0

4. `yarn --version` (if you use Yarn):
1.9.4

Then, specify:

1. Operating system:
macOS High Sierra

### Steps to Reproduce

1. yarn create react-app my-app --scripts-version=react-scripts-ts
2. create an enum in the source directory
```ts
export const enum MyConstEnum {
  One = 'One',
  Two = 'Two'
}
```
3. 
Import the enum in app and only use 'One'.

```ts
import * as React from 'react';
import { MyConstEnum } from './MyConstEnum';

class App extends React.Component {
  public render() {
    return (
      MyConstEnum.One
    );
  }
}

export default App;
```
4. yarn build
5. Inspect build/static/js/main*.js seach for "One" and we find
function(e){e.One="One",e.Two="Two"}(r||(r={}))}
And 
t.prototype.render=function(){return i.a.One}
instead of the inlined version

### Expected Behavior

If we manually changes the transpile flag to false and run yarn build again and inspects main.js

t.prototype.render=function(){return"One"}

And we got the inlined version :) 

### Actual Behavior

We uses the non inlined version (as described in step 5)

### Reproducible Demo

Steps to Reproduce should be enough, otherwise I can push a demo project.

**Comment by axelo** [comment]
We also uses const enums for redux-actions so some bytes could be saved for other projects (no data on this though).  But maybe this isn't that big of a problem for most projects, and maybe some information about the trade off, and if possible, an optional flag, with the effect of slowing down the build a bit.

**Comment by evaera** [comment]
Note that `const enum`s may appear in declaration files. Some modules may accept either literal strings or (even worse) special numbers with significant meaning. When writing type declarations, creating `const enum`s for these situations is ideal because you get proper tooling while still being able to be compatible with the source library.

Because of that, if these enums are not inlined properly (even in development mode!) the transpilation step will produce a type error, as that enum does not exist as a value anywhere in the code.

**Comment by DorianGrey** [comment]
See my description above about the reason.

Still thinking about a good way to disable `transpileOnly` without doing duplicate work via the type checker, and without losing its linting capabilites (there is no immediate replacement for that atm.).

**Comment by evaera** [comment]
Yeah, I understand the *technical* reason, I just mean that there is no obvious reason as to why it doesn't work from a UX standpoint, especially because it results in a type error.

**Comment by ptommasi** [comment]
I had a similar issue (an enum inside a declared but not exported namespace was undefined once transpiled), I [asked on stackoverflow](https://stackoverflow.com/questions/52817342/access-an-enum-in-a-declared-and-not-exported-module-in-typescript-once-transp) and the `const enum` would have been my preferred option. 

In my case, I exported the namespace (rather than declaring it) and exported the enum inside, as expected the generated code didn't have the inline value but at least the variable wasn't undefined anymore and the UI is working. 

As a temporary fix, I exported the namespace with all the declaration (enum included) and I have to manually import it in most of my modules. That import line in every file is a bit annoying, but at least it's not breaking... Don't know if it helps someone.
